File: solutions/cpp/secret-handshake/1/secret_handshake.cpp
#include "secret_handshake.h"

namespace secret_handshake {
    auto commands(uint8_t number) -> std::vector<std::string> {
        const auto actions = std::array<std::string, 4> {
            "wink",
            "double blink",
            "close your eyes",
            "jump"
        };

        auto sequence = std::vector<std::string>{};

        for (uint8_t shift{}; shift < 4; shift++) {
            if (number bitand (1 << shift)) {
                sequence.emplace_back(actions.at(shift));
            }
        }

        if (number bitand 16) {
            std::reverse(sequence.begin(), sequence.end());
        }

        return sequence;
    }
}  // namespace secret_handshake

File: solutions/javascript/secret-handshake/1/secret-handshake.js
const actionsList = [
  "wink",
  "double blink",
  "close your eyes",
  "jump",
];

export const commands = (number) => {
  const list = actionsList.filter((_, idx) => number & (1 << idx));
  return number & (1 << 4) ? list.reverse() : list;
};
